vpr/src/place/timing_place.cpp

```diff
@@ -100,6 +107,77 @@ PlacerCriticalities::pin_range PlacerCriticalities::pins_with_modified_criticali
     return vtr::make_range(cluster_pins_with_modified_criticality_);
 }
 
```

Comment:
Are some of these routines rewrites / replacements for routines in place.cpp?  I haven't checked line by line but some seem similar.